test/integration/index.spec.js

```diff
@@ -49,118 +83,208 @@ describe(`integration with ${authType}`, function () {
-  });
-
-  it('should create a new session for new users', async () => {
-    const auth = { username: tempAdminName, password: 'spacesareaproblem' };
-    await utils.createAdmin(auth.username, auth.password);
-    await utils.createDb(tempDbName);
-
-    tempDb = getDb(tempDbName, auth, authType);
-
-    const collectLogs = await utils.getDockerContainerLogs();
-    await db.allDocs();
-    await tempDb.allDocs();
-
-    const logs = await collectLogs();
-
-    expect(utils.getDbRequest(auth.username, logs, tempDbName, '/_all_docs').length).to.equal(1);
-    expect(utils.getSessionRequests(logs).length).to.equal(1);
-    expect(utils.getCookieAuthRequests(auth.username, logs).length).to.equal(1);
-  });
-
-  it('should throw errors on password changes', async () => {
-    const auth = { username: tempAdminName, password: 'new_password' };
-    await utils.createAdmin(auth.username, auth.password);
-    await utils.createDb(tempDbName);
-
-    tempDb = getDb(tempDbName, auth, authType);
-
-    const collectLogs = await utils.getDockerContainerLogs();
-    await tempDb.allDocs();
-    await utils.createAdmin(auth.username, 'password change');
-    await expect(tempDb.allDocs()).to.eventually.be.rejectedWith('Name or password is incorrect.');
-    const logs = await collectLogs();
-
-    expect(utils.getSessionRequests(logs, false).length).to.equal(1);
-    expect(utils.getDbRequest('undefined', logs, tempDbName, '/_all_docs', false).length).to.equal(2);
-  });
-
-  it('should throw errors on invalid credentials', async () => {
-    const newDb = getDb(dbName, { username: utils.dbAuth.username, password: 'not the right password' }, authType);
-
-    const collectLogs = await utils.getDockerContainerLogs();
-    await expect(newDb.allDocs()).to.eventually.be.rejectedWith('Name or password is incorrect.');
-    const logs = await collectLogs();
-    expect(utils.getSessionRequests(logs, false).length).to.equal(1);
-  });
-
-  it('should automatically refresh the cookie when closing to expiry', async () => {
-    await utils.asyncTimeout(3000);
-    await db.allDocs();
-    await utils.asyncTimeout(3000);
-    await db.allDocs();
-  });
-
-  it('should automatically refresh the cookie when expired', async () => {
-    const collectLogs = await utils.getDockerContainerLogs();
-    await utils.asyncTimeout(6000);
-    await db.allDocs();
-    const logs = await collectLogs(1000);
-    expect(utils.getSessionRequests(logs).length).to.equal(1);
-  });
-
-  it('should support initial setup', async () => {
-    const auth = { username: tempAdminName, password: 'new_password' };
-    await utils.createAdmin(auth.username, auth.password);
-
-    const collectLogs = await utils.getDockerContainerLogs();
-    tempDb = getDb(tempDbName, auth, authType, false);
-    await tempDb.allDocs();
-    const logs = await collectLogs(1000);
-    expect(utils.getSessionRequests(logs).length).to.equal(1);
+  describe('#nosession auth type', () => {
```

Comment:
I'm finding it difficult to figure out what tests will run based on what params I provide.

Instead of using hashtags and greps, I think it would be cleaner to check the environment variable in the test, eg:

```suggestion
  describe('auth type', () => {
    if (authType === 'session') {
      this.skip();
    }
```

This way you only have one variable to set (the AUTH_TYPE env var) rather than the grep value as well, and you can't get into invalid situations like the wrong grep with the wrong auth type.

Alternatively if there was a way to split the test files with reuse that could work too.

Reply:
This part - deciding how to split these tests - took me the longest. I was just not happy with any solution. 
I'll split them in separate files.

Reply:
I've split the tests.